sstables/sstables.cc

```diff
@@ -1284,7 +1302,9 @@ future<> sstable::update_info_for_opened_data() {
     }).then([this] {
         this->set_clustering_components_ranges();
         this->set_first_and_last_keys();
-        _run_identifier = _components->scylla_metadata->get_optional_run_identifier().value_or(utils::make_random_uuid());
+        if (_scylla_metadata) {
+            _run_identifier = _scylla_metadata->get_optional_run_identifier().value_or(utils::make_random_uuid());
```

Comment:
Should we even have a _scylla_metadata member? if we can get rid of it we can be sure we don't have some code path that keeps is around.

Reply:
Yes, agree that it's not easy to get rid of it for the write path. One day we'll move it to the sstable writer class.